go/enclave/evm/evm_facade.go

```diff
@@ -110,3 +130,53 @@ func secret(storage db.Storage) []byte {
 	secret := storage.FetchSecret()
 	return secret[:]
 }
+
+func newErrorWithReasonAndCode(err error) SerialisableError {
+	result := SerialisableError{
+		Err: err.Error(),
+	}
+
+	var e gethrpc.Error
+	ok := errors.As(err, &e)
+	if ok {
+		result.Code = e.ErrorCode()
+	}
+	var de gethrpc.DataError
+	ok = errors.As(err, &de)
+	if ok {
+		result.Reason = de.ErrorData()
+	}
+	return result
+}
+
+func newRevertError(result *gethcore.ExecutionResult) SerialisableError {
+	reason, errUnpack := abi.UnpackRevert(result.Revert())
+	err := errors.New("execution reverted")
+	if errUnpack == nil {
+		err = fmt.Errorf("execution reverted: %v", reason)
+	}
+	return SerialisableError{
+		Err:    err.Error(),
+		Reason: hexutil.Encode(result.Revert()),
+		Code:   3, // todo - magic number
+	}
+}
+
+// SerialisableError is an API error that encompasses an EVM error with a code
+type SerialisableError struct {
+	Err    string
+	Reason interface{}
+	Code   int
+}
+
+func (e SerialisableError) Error() string {
+	return e.Err
+}
+
+func (e SerialisableError) ErrorCode() int {
```

Comment:
This can be removed, I think.

Reply:
it's needed to implement some interfaces from the geth rpc package